packages/pagination/src/Pagination.js

```diff
@@ -94,6 +95,17 @@ const Pagination = ({
     }
   };
 
+  // We don't want to reset the page on the first render
+  const firstUpdate = useRef(true);
+  useEffect(() => {
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+    } else {
+      updatePage(1);
```

Comment:
This won't work in the case the user is on the 1st page and some resetParam prop changes when the `items` prop is a function.

Reply:
can you elaborate? i'm not following 😄

Reply:
For the below example if I am on page 1 of the `PaginationComponent` and I update say the `hello` prop then it will not force the re-render.


```jsx
<Pagination
items={() => {
 fetch(`/some-api/${id}`,{
 filters: ...someFilters
})
}}
watchList={[...someFilters]}
resetParams={[id]}
>
 ... Components
</Pagination>
```

Reply:
isn't that what `watchList` is for? So, for this example, `id` should be part of your `watchList` ?

Reply:
3823a95a1bb0288049af593e9a20bfad43a3a353

Reply:
I think we should modify the test to test for the items being rendered properly as well. That test doesn't solidify the fact the component does call the render method properly.

Reply:
f3433b5e5e52821607fcf612b2d9e5c91a094602